### DISCUSSION TITLE
Password reset security issue

### DISCUSSION THREAD

**Discussion by vienfla** [danny-avila/LibreChat#1653] [Troubleshooting] [answered]
### What happened?

Hello!
On the /forgot-password page, when a user enters an existing email account, the reset link containing the secret token is displayed directly on the page. 
This behavior could allow anyone to reset the password of any account without requiring access to the associated email account.

Many thanks!

### Steps to Reproduce

1. go to the login page
2. click on "Forgot Password?"
3. enter email address
4. Click on [HERE] to reset your password
5. reset the password without any email verification.

### Screenshots

![Screenshot from 2024-01-27 01-47-12](https://github.com/danny-avila/LibreChat/assets/53423096/d654bfdc-ec96-4189-8118-bae42d638f03)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment, marked as answer]
Hi, thanks for your concern. This is not a bug, and is the default behavior when an Email service is not configured.

https://docs.librechat.ai/install/configuration/user_auth_system.html#email-and-password-reset

From the docs:

> NOTE: ⚠️ Failing to perform either of the below setups will result in LibreChat using the unsecured password reset! This allows anyone to reset any password on your server immediately, without mail being sent at all! The variable EMAIL_FROM does not support all email providers but is still required . To stay updated, check the bug fixes: here

**Reply by vienfla** [reply]
Thank you for your assistance. 
I am using a local postfix server for sending emails (which does not require authentication), so most EMAIL_ variables are not set. However, I discovered that by assigning fake values to EMAIL_USERNAME and EMAIL_PASSWORD, instead of leaving them empty, I was able to successfully initiate the password reset functionality that I expected.

EMAIL_SERVICE=
EMAIL_HOST= localhost                    
EMAIL_PORT=25                   
EMAIL_ENCRYPTION=               
EMAIL_ENCRYPTION_HOSTNAME=      
EMAIL_ALLOW_SELFSIGNED=         
EMAIL_USERNAME= fake            
EMAIL_PASSWORD=  fake           
EMAIL_FROM_NAME= LibreChat           
EMAIL_FROM=noreply@mydomain.com

**Comment by Kevin-free** [comment]
@vienfla  @danny-avila  Thank you for the information! I tried adding the `Email Password Reset` environment variable and was able to successfully send a password reset email.

Now I want to know **how to send a verification code email when registering?**

Is there a direct configuration that can be modified? Or do I need to add code to achieve it? Thank you!

**Reply by danny-avila** [reply]
@Kevin-free There is an open PR for this that I hope to complete this week

**Reply by Kevin-free** [reply]
Awesome! This is a great project, I've added some features as well and hope to have the opportunity to contribute.
